tool/scaffold/src/main/java/org/openscience/cdk/tools/scaffold/ScaffoldGenerator.java

```diff
@@ -0,0 +1,2412 @@
+         * All terminal side chains are removed and only linkers and rings are retained.
+         */
+        MURCKO_FRAMEWORK(),
+
+        /**
+         * All side chains are removed, all bonds are converted into single bonds and all atoms are converted into carbons.
+         * Naming is based on <a href="https://doi.org/10.1186/s13321-021-00526-y">
+         * "Molecular Anatomy: a new multi‑dimensional hierarchical scaffold analysis tool"</a>
+         * by Manelfi et al. 2021.
+         */
+        BASIC_WIRE_FRAME(),
+
+        /**
+         * All side chains are removed and multiple bonds are converted to single bonds, but the atomic elements remain.
+         */
+        ELEMENTAL_WIRE_FRAME(),
+
+        /**
+         * All side chains are removed and all atoms are converted into carbons. The order of the remaining bonds is not changed.
+         * Naming is based on <a href="https://doi.org/10.1186/s13321-021-00526-y">
+         * "Molecular Anatomy: a new multi‑dimensional hierarchical scaffold analysis tool"</a>
+         * by Manelfi et al. 2021.
+         */
+        BASIC_FRAMEWORK()
+    }
+    //</editor-fold>
+
+    //<editor-fold desc="Public static final constants">
+    /**
+     * Property of the atoms according to which they are counted and identified.
+     */
+    public static final String SCAFFOLD_ATOM_COUNTER_PROPERTY = "SCAFFOLD_ATOM_COUNTER_PROPERTY";
+
+    /**
+     * Cycle finder used to detect rings.
+     */
+    public static final CycleFinder CYCLE_FINDER = Cycles.relevant();
+
+    /**
+     * Property is true if the backup cycle finder is to be used instead of the normal cycle finder.
+     */
+    public static final String CYCLE_FINDER_BACKUP_PROPERTY = "CYCLE_FINDER_BACKUP_PROPERTY";
+
+    /**
+     * Backup cycle finder used to detect rings.
+     * The relevant cycle finder has problems with a few molecules and also finds too many rings in some molecules.
+     * Therefore, the mcb is used in these cases.
+     */
+    public static final CycleFinder CYCLE_FINDER_BACKUP = Cycles.mcb();
+
+    /**
+     * Default setting for whether the aromaticity should be determined.
+     * By default, the aromaticity is determined.
+     */
+    public static final boolean DETERMINE_AROMATICITY_SETTING_DEFAULT = true;
+
+    /**
+     * Default setting for which aromaticity model should be used.
+     * By default, Aromaticity(ElectronDonation.cdk(), Cycles.cdkAromaticSet()) is used.
+     */
+    public static final Aromaticity AROMATICITY_MODEL_SETTING_DEFAULT = new Aromaticity(ElectronDonation.cdk(), Cycles.cdkAromaticSet());
+
+    /**
+     * Default setting for which SmilesGenerator should be used.
+     * By default, unique SMILES are used.
+     */
+    public static final SmilesGenerator SMILES_GENERATOR_SETTING_DEFAULT = new SmilesGenerator(SmiFlavor.Unique | SmiFlavor.UseAromaticSymbols);
+
+    /**
+     * Default setting for whether rule 7 should be applied. By default, rule 7 is applied.
```

Comment:
Rule 7 of what?

Reply:
I have added more details in the latest commit.